### PR TITLE
clean random gentxs and add better validation

### PR DESCRIPTION
**Pull request by khoslaventures**

### CODE REVIEW

**Comment by khoslaventures** [comment]
```
whitespace on /Users/akash/testnets/olympus_mons/gentx-300/Create Ankaria.json; won't process
whitespace on /Users/akash/testnets/olympus_mons/gentx-300/Add fible1.json; won't process
status check (faulty params or signatures, panic: signature verification failed; please verify account number (0) and chain-id (evmos_9000-2): unauthorized) failed on /Users/akash/testnets/olympus_mons/gentx-300/sleepykitten.json
status check (faulty params or signatures, panic: empty description: invalid request) failed on /Users/akash/testnets/olympus_mons/gentx-300/TEOLIDER.json
bonded too much: 10000000000000 > 1000000000000 on /Users/akash/testnets/olympus_mons/gentx-300/sergioch.json
status check (faulty params or signatures, panic: commission cannot be more than 100%) failed on /Users/akash/testnets/olympus_mons/gentx-300/Wetez.json
status check (faulty params or signatures, panic: signature verification failed; please verify account number (0) and chain-id (evmos_9000-2): unauthorized) failed on /Users/akash/testnets/olympus_mons/gentx-300/yatsvic.json
status check (faulty params or signatures, panic: signature verification failed; please verify account number (0) and chain-id (evmos_9000-2): unauthorized) failed on /Users/akash/testnets/olympus_mons/gentx-300/devops_mne.json
status check (faulty params or signatures, panic: pubKey does not match signer address evmos1hsxs04fvxcsqjstvnkvu6p0umfy86lzmsd2hhs with signer index: 0: invalid pubkey) failed on /Users/akash/testnets/olympus_mons/gentx-300/ArtiDS_Node.json
status check (faulty params or signatures, panic: commission cannot be more than 100%) failed on /Users/akash/testnets/olympus_mons/gentx-300/pospool.json
status check (faulty params or signatures, panic: signature verification failed; please verify account number (0) and chain-id (evmos_9000-2): unauthorized) failed on /Users/akash/testnets/olympus_mons/gentx-300/simply-staking.json
collect-gentxs failed on /Users/akash/testnets/olympus_mons/gentx-300/tophata.json
status check (faulty params or signatures, panic: signature verification failed; please verify account number (0) and chain-id (evmos_9000-2): unauthorized) failed on /Users/akash/testnets/olympus_mons/gentx-300/softstake.json
```

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+./scripts/validate-gentx-macos.sh
+```
+
+You may need to adjust env variables. Here are the defaults
+```
+EVMOS_HOME="/tmp/evmosd$(date +%s)"
+RANDOM_KEY="randomevmosvalidatorkey"
+CHAIN_ID="evmos_9000-2"
+DENOM="aphoton"
+MAXBOND="1000000000000" # 1 PHOTON
```
Comment (fedekunze) [inline scripts/README.md:29]: this is 1 Photon:
```suggestion
MAXBOND="1000000000000000000" # 1 PHOTON
```

File: scripts/make-genesis.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+EVMOS_HOME="/tmp/evmosd$(date +%s)"
+CHAIN_ID="evmos_9000-2"
+DENOM="aphoton"
+MAXBOND="1000000000000" # 1 PHOTON
```
Comment (fedekunze) [inline scripts/make-genesis.sh:5]:
```suggestion
MAXBOND="1000000000000000000" # 1 PHOTON
```

File: scripts/make-genesis.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+DAEMON="./build/evmosd"
+GH_URL="https://github.com/tharsis/evmos"
+BINARY_VERSION="v0.2.0"
+GENTXS_DIR="$HOME/testnets/olympus_mons/valid-gentxs"
+FAUCET1="evmos1ht560g3pp729z86s2q6gy5ws6ugnut8r4uhyth"
+FAUCET2="evmos1hefvrgzc85hmn2nwdk3lhttk6jwlzzgv6e8tmc"
+FAUCET_BALANCE="250000000000000000000000"
```
Comment (fedekunze) [inline scripts/make-genesis.sh:12]:
```suggestion
FAUCET_BALANCE="250000000000000000000000" # 250k PHOTONS
```

File: scripts/validate-gentx.sh [hunk 4]
```diff
@@ -0,0 +1,160 @@
+#!/bin/sh
+EVMOS_HOME="/tmp/evmosd$(date +%s)"
+RANDOM_KEY="randomevmosvalidatorkey"
+CHAIN_ID="evmos_9000-2"
+DENOM="aphoton"
+MAXBOND="1000000000000" # 1 PHOTON
```
Comment (fedekunze) [inline scripts/validate-gentx.sh:6]:
```suggestion
MAXBOND="1000000000000000000" # 1 PHOTON
```

File: scripts/validate-gentxs-macos.sh [hunk 5]
```diff
@@ -0,0 +1,157 @@
+#!/bin/sh
+EVMOS_HOME="/tmp/evmosd$(date +%s)"
+RANDOM_KEY="randomevmosvalidatorkey"
+CHAIN_ID="evmos_9000-2"
+DENOM="aphoton"
+MAXBOND="1000000000000" # 1 PHOTON
```
Comment (fedekunze) [inline scripts/validate-gentxs-macos.sh:6]:
```suggestion
MAXBOND="1000000000000000000" # 1 PHOTON
```

File: scripts/validate-gentxs-macos.sh [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    # check for duplicate accounts
+    OUTPUT=$($DAEMON add-genesis-account "$GENACC" 1000000000000000$DENOM --home "$EVMOS_HOME" 2>&1 || true)
+    if contains "$OUTPUT" "Error"; then
+        echo "add-genesis-account failed on $GENTX_FILE" | tee -a "$OUTFILE"
+        # remove gentxs and reset genesis
+        rm -rf "$EVMOS_HOME" >/dev/null 2>&1
+        continue
+    fi 
+
+    $DAEMON add-genesis-account $RANDOM_KEY 100000000000000$DENOM --home "$EVMOS_HOME" \
```
Comment (fedekunze) [inline scripts/validate-gentxs-macos.sh:113]: I think you also need to update the amount here

**Pull request merged**